From 0904fd4b3fdbc99f8cb61d325c71c4f72682599e Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 3 Jun 2018 17:25:10 +0300
Subject: [PATCH 0643/1051] dts: arm64: marvell: add SFP nodes to a8kp device
 trees

This patch fix 10G interface start fail issue due to missed
SFP nodes required by phylink if SFP cage used.

Change-Id: Ic84befa8e0fb7ce407d8a5e6a8b35ec6e19280e5
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../boot/dts/marvell/armada-8082-db-B.dts     | 12 ++++++++++
 .../arm64/boot/dts/marvell/armada-8082-db.dts | 12 ++++++++++
 .../arm64/boot/dts/marvell/armada-8084-db.dts | 24 +++++++++++++++++++
 3 files changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts
index 6ff08608e948..fad6c46b402d 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db-B.dts
@@ -76,6 +76,14 @@
 		ethernet2 = &cp0_eth2;
 		ethernet3 = &cp1_eth0;
 	};
+
+	sfp_eth0: sfp-eth0 {
+		compatible = "sff,sfp";
+	};
+
+	sfp_eth1: sfp-eth1 {
+		compatible = "sff,sfp";
+	};
 };
 
 &ap0_sdhci0 {
@@ -152,6 +160,8 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth0>;
 };
 
 &cp0_eth1 {
@@ -205,4 +215,6 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp1_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth1>;
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db.dts b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
index 137ba52df043..db744a0c9d47 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
@@ -67,6 +67,14 @@
 		ethernet3 = &cp1_eth0;
 		spi0 = &ap0_spi0;
 	};
+
+	sfp_eth0: sfp-eth0 {
+		compatible = "sff,sfp";
+	};
+
+	sfp_eth1: sfp-eth1 {
+		compatible = "sff,sfp";
+	};
 };
 
 &ap0_spi0 {
@@ -143,6 +151,8 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth0>;
 };
 
 &cp0_eth1 {
@@ -196,4 +206,6 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp1_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth1>;
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8084-db.dts b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
index f9e665d107de..16bab0e6d81e 100644
--- a/arch/arm64/boot/dts/marvell/armada-8084-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
@@ -69,6 +69,22 @@
 		ethernet5 = &cp3_eth0;
 		spi0 = &ap0_spi0;
 	};
+
+	sfp_eth0: sfp-eth0 {
+		compatible = "sff,sfp";
+	};
+
+	sfp_eth1: sfp-eth1 {
+		compatible = "sff,sfp";
+	};
+
+	sfp_eth2: sfp-eth2 {
+		compatible = "sff,sfp";
+	};
+
+	sfp_eth3: sfp-eth3 {
+		compatible = "sff,sfp";
+	};
 };
 
 &ap0_spi0 {
@@ -153,6 +169,8 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth0>;
 };
 
 &cp0_eth1 {
@@ -206,6 +224,8 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp1_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth1>;
 };
 
 &cp2_sata0 {
@@ -239,6 +259,8 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp2_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth2>;
 };
 
 &cp3_sata0 {
@@ -272,4 +294,6 @@
 	phy-mode = "10gbase-kr";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp3_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth3>;
 };
-- 
2.17.1

